interface/patient_file/summary/demographics.php

```diff
@@ -749,78 +751,110 @@ function setMyPatient() {
 if ($GLOBALS['patient_id_category_name']) {
     $idcard_doc_id = get_document_by_catg($pid, $GLOBALS['patient_id_category_name']);
 }
+$res=sqlStatement("SELECT option_id,activity FROM list_options WHERE list_id='patient_file_menu' ");
+
+while ($row = sqlFetchArray($res)) {
+    $patient_file_menu[$row['option_id']]=$row['activity'];
+}
+
+// Use a json file to build menu
+// Collect the selected menu of user
+$patientMenuRole = PatientMenuRole::getPatientMenuRole();
+// Load the selected menu
+if (preg_match("/.json$/", $patientMenuRole)) {
+    // load custom menu (includes .json in id)
+    $menu_parsed = json_decode(file_get_contents($GLOBALS['OE_SITE_DIR'] . "/documents/custom_menus/patient/menus/" . $patientMenuRole));
+} else {
+    // load a standardized menu (does not include .json in id)
+    $menu_parsed = json_decode(file_get_contents($GLOBALS['fileroot'] . "/interface/main/tabs/menu/menus/patient_menus/" . $patientMenuRole . ".json"), true);
+}
+// if error, then die and report error
+if (!$menu_parsed) {
+    die("\nJSON ERROR: " . json_last_error());
+}
+menu_update_entries($menu_parsed);
+$menu_restrictions=array();
+//menu_apply_restrictions($menu_parsed, $menu_restrictions, $menu_type);
```

Comment:
Note you will work in the filtering into menu_apply_restrictions()(this is why use global_req since it is already built into the function). Then at that point your $menu_parsed will contain only the elements you need to make the menu (thus no filtering will be needed after that).